Sources/SwiftGenKit/Stencil/Files+Context.swift

```diff
@@ -0,0 +1,60 @@
+//
+// SwiftGenKit
+// Copyright © 2020 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+import PathKit
+
+extension Files.Parser {
+  public func stencilContext() -> [String: Any] {
+    let files = self.files
+      .sorted { lhs, rhs in lhs.name < rhs.name }
+
+    return structure(entries: files, usingMapper: map(file:))
+  }
+
+  private func map(file: Files.File) -> [String: Any] {
+    [
+      "name": file.name,
+      "ext": file.ext ?? "",
+      "path": file.path.joined(separator: Path.separator),
+      "mimeType": file.mimeType
+    ]
+  }
+
+  typealias Mapper = (_ file: Files.File) -> [String: Any]
+  private func structure(
+    entries: [Files.File],
+    atKeyPath keyPath: [String] = [],
+    usingMapper mapper: @escaping Mapper
+  ) -> [String: Any] {
+    var structuredFiles: [String: Any] = [:]
+    if let name = keyPath.last {
+      structuredFiles["name"] = name
+    }
+
+    let files = entries
```

Comment:
We should add the same comments as in the strings context, this function can get confusing otherwise.